src/TikaOnDotnet.TextExtractor/TextExtractor.cs

```diff
@@ -1,137 +1,241 @@
+
+            var configDetector = (CompositeDetector)Config.getDetector();
+            var detectors = configDetector.getDetectors().toArray();
+            foreach (Detector detector in detectors)
+            {
+                retVal.AppendFormat("\t{0}\n", ((Object)detector).getClass().getName());
+
+                if (detector.GetType() == typeof(CompositeDetector))
+                {
+                    var subDetectors = configDetector.getDetectors().toArray();
+                    foreach (Detector subDetector in subDetectors)
+                    {
+                        retVal.AppendFormat("\t\t{0}\n", ((Object)subDetector).getClass().getName());
+                    }
+                }
+            }
+
+            retVal.AppendLine("\nParsers");
+
+            var configParser = (CompositeParser)Config.getParser();
+            var parsers = configParser.getAllComponentParsers().toArray();
+            foreach (Parser parser in parsers)
+            {
+                retVal.AppendFormat("\t{0}\n", ((Object)parser).getClass().getName());
+
+                var parserTypes = parser.getSupportedTypes(new ParseContext()).toArray();
+                foreach (MediaType mediaType in parserTypes)
+                {
+                    retVal.AppendFormat("\t\t{0}\n", mediaType.toString());
+                }
+            }
+
+            var translator = Config.getTranslator();
+            if (translator.isAvailable())
+            {
+                retVal.AppendFormat("Translator {0}\n", ((Object)translator).getClass().getName());
+            }
+
+            retVal.AppendFormat("\nFallback Parser: {0}\n", configParser.getFallback());
+
+            return retVal.ToString();
+        }
+        #endregion
+
+        #region AssembleExtractionResult
+        private static TextExtractionResult AssembleExtractionResult(string text, Metadata metadata)
+        {
+            var metaDataResult = metadata.names()
+                .ToDictionary(name => name, name => string.Join(", ", metadata.getValues(name)));
+
+            var contentType = metaDataResult["Content-Type"];
+
+            return new TextExtractionResult
+            {
+                Text = text,
+                ContentType = contentType,
+                Metadata = metaDataResult
+            };
+        }
+        #endregion
+
+        #region GetTransformerHandler
+        internal static TransformerHandler GetTransformerHandler(Writer output)
+        {
+            var factory = (SAXTransformerFactory)TransformerFactory.newInstance();
+            var transformerHandler = factory.newTransformerHandler();
+
+            transformerHandler.getTransformer().setOutputProperty(OutputKeys.METHOD, "text");
+            transformerHandler.getTransformer().setOutputProperty(OutputKeys.INDENT, "yes");
+
+            transformerHandler.setResult(new StreamResult(output));
+            return transformerHandler;
+        }
+        #endregion
+    }
 }
```

Comment:
You might not be to blame for the line endings now that I think of it. They were likely like that before.